packages/auth-construct/src/construct.ts

```diff
@@ -423,8 +458,41 @@ export class AmplifyAuth
       standardAttributes: {
         email: DEFAULTS.IS_REQUIRED_ATTRIBUTE.email(emailEnabled),
         phoneNumber: DEFAULTS.IS_REQUIRED_ATTRIBUTE.phoneNumber(phoneEnabled),
-        ...(props.userAttributes ? props.userAttributes : {}),
+        ...(props.userAttributes
+          ? Object.entries(props.userAttributes).reduce(
```

Comment:
Instead of looping `Object.entries(props.userAttributes)` twice on line 462 and line 475, we can loop it once and get two arrays. 
Something like `{ userAttributes, customAttibutes } = Object.entries(props.userAttributes).reduce(...)

Reply:
Updated.